pmc/journal_recommendation/journal_recommendation_figure_generator.py

```diff
@@ -0,0 +1,570 @@
+    p9.ggplot(plot_df)
+    + p9.aes(x="dim1", y="dim2", fill = "journal")
+    + p9.facet_grid("lambda_d ~ lambda_c", labeller="label_both", scales="free")
+    + p9.geom_point()
+    + p9.theme(
+        figure_size=(12, 12)
+    )
+)
+print(g)
+
+
+# In[14]:
+
+
+lambda_c_grid = np.linspace(1e-6, 1e-3, num=5)
+lambda_d_grid = np.linspace(1e-6, 1e-3, num=5)
+
+
+# In[15]:
+
+
+plot_df = run_saucie_param_grid(
+    full_paper_dataset_subset,
+    lambda_c_grid=lambda_c_grid,
+    lambda_d_grid=lambda_d_grid
+)
+
+
+# In[16]:
+
+
+g = (
+    p9.ggplot(plot_df)
+    + p9.aes(x="dim1", y="dim2", fill = "journal")
+    + p9.facet_grid(
+        "lambda_d ~ lambda_c", 
+        labeller=p9.labeller(
+            cols=lambda s:f"lambda_c: {float(s):.3e}",
+            rows=lambda s:f"lambda_d: {float(s):.3e}"
+        ),
+        scales="free"
+    )
+    + p9.geom_point()
+    + p9.theme(
+        figure_size=(12, 12)
+    )
+)
+g.save("output/figures/saucie_hyperparam_lambda_cd.png", dpi=500)
+print(g)
+
+
+# In[17]:
+
+
+learning_rate_grid = np.linspace(1e-6, 1e-3, num=3)
+steps_grid = [1000, 3000, 5000, 10000, 10000]
+
+
+# In[18]:
+
+
+plot_df = run_saucie_param_grid(
+    full_paper_dataset_subset,
+    lambda_c_grid=[1.000e-3],
+    lambda_d_grid=[1.000e-3], 
+    steps_grid=steps_grid,
+    learning_rate_grid=learning_rate_grid
+)
+
+
+# In[19]:
+
+
+g = (
+    p9.ggplot(plot_df)
+    + p9.aes(x="dim1", y="dim2", fill = "journal")
+    + p9.facet_grid(
+        "steps ~ learning_rate", 
+        labeller=p9.labeller(
+            cols=lambda s:f"learning_rate: {float(s):.3e}",
+            rows=lambda s:f"steps: {s}"
+        ),
+        scales="free"
+    )
+    + p9.geom_point()
+    #+ p9.scale_fill_discrete(guide=False)
+    + p9.theme(
+        figure_size=(12, 12)
+    )
+)
+g.save("output/figures/saucie_hyperparam_lr_steps.png", dpi=500)
+print(g)
+
+
+# In[20]:
+
+
+set_seeds()
+
+saucie = SAUCIE(
+    full_paper_dataset.shape[1]-2, 
+    lambda_b=0, lambda_c=1e-3, lambda_d=1e-3,
+    learning_rate=1e-3,
+    save_folder="output/model"
+)
+
+loadtrain = Loader(
+    full_paper_dataset.drop(["journal", "document"], axis=1).values, 
+    pd.Categorical(full_paper_dataset["journal"].values).codes, 
+    shuffle=True
+)
+
+saucie.train(loadtrain, steps=10000)
+saucie.save()
+
+loadeval = Loader(
+    full_paper_dataset.drop(["journal", "document"], axis=1).values, 
+    pd.Categorical(full_paper_dataset["journal"].values).codes, 
+    shuffle=False
+)
+embedding = saucie.get_embedding(loadeval)
+embedding
+
+
+# In[21]:
+
+
+full_dataset = (
+    pd.DataFrame(
+        embedding[0], 
+        columns=["dim1", "dim2"]
+    )
+    .assign(
+        journal = full_paper_dataset.journal.tolist(),
+        document = full_paper_dataset.document.tolist()
+    )
+)
+
+full_dataset.to_csv(
+    Path("output/paper_dataset")/
+    Path("paper_dataset_full_tsne.tsv"),
+    sep="\t", index=False
+)
+
+full_dataset.head()
+
+
+# In[22]:
+
+
+g = (
```

Comment:
Since this plot is going in your app, maybe rename the axes to be more interpretable..?

Reply:
>Since this plot is going in your app, maybe rename the axes to be more interpretable..?

For the notebook this wont be necessary as the journal recommendation app doesn't require axis labels. The dimension names here are pretty arbitrary as the axes don't really have an interpretable meaning. Good point for future experiments to keep in mind.